src/gpg_fmt_plug.c

```diff
@@ -402,29 +413,37 @@ static int valid(char *ciphertext, struct fmt_main *self)
 	} else if (usage == 255 && spec == 3 && algorithm == 1) {
 		/* gpg --homedir . --s2k-cipher-algo 3des --simple-sk-checksum --gen-key */
 		ex_flds = 1; /* handle p */
+	} else if (symmetric_mode) {
+		ex_flds = 1; /* handle the hash appended at the end */
 	} else {
 		/* NOT sure what to do here, probably nothing */
 	}
 
 	p = strtokm(NULL, "*"); /* NOTE, do not goto err if null, we WANT p nul if there are no fields */
 
-	for (j = 0; j < ex_flds; ++j) {  /* handle extra p, q, g, y fields */
-		if (!p) /* check for null p */
-			goto err;
-		if (!isdec(p))
-			goto err;
-		res = atoi(p);
-		if (res > BIG_ENOUGH * 2)
-			goto err; // FIXME: warn if BIG_ENOUGH isn't big enough?
-		if ((p = strtokm(NULL, "*")) == NULL)
-			goto err;
-		if (hexlenl(p) != res*2)
+	if (!symmetric_mode) {
+		for (j = 0; j < ex_flds; ++j) {  /* handle extra p, q, g, y fields */
+			if (!p) /* check for null p */
+				goto err;
+			if (!isdec(p))
+				goto err;
+			res = atoi(p);
+			if (res > BIG_ENOUGH * 2)
+				goto err; // FIXME: warn if BIG_ENOUGH isn't big enough?
+			if ((p = strtokm(NULL, "*")) == NULL)
+				goto err;
+			if (hexlenl(p) != res*2)
+				goto err;
+			p = strtokm(NULL, "*");  /* NOTE, do not goto err if null, we WANT p nul if there are no fields */
+		}
+		if (p)	/* at this point, there should be NO trailing stuff left from the hash. */
 			goto err;
-		p = strtokm(NULL, "*");  /* NOTE, do not goto err if null, we WANT p nul if there are no fields */
 	}
 
-	if (p)	/* at this point, there should be NO trailing stuff left from the hash. */
-		goto err;
+	if (symmetric_mode) {
+		// fprintf(stderr, "GPG symmetric mode support is in progress!\n");
+		// goto err;
+	}
```

Comment:
You need to clean up all these weird artifacts before merging. This is an empty body. Also, lots of commented-out stuff should actually be removed once it's working.